### ISSUE TITLE
[QUESTION]How to use Chinese Key Vault

### ISSUE THREAD

**Issue by sunmsy** [microsoft/mssql-jdbc#1126]
When I use the Chinese key vault, I get the following error
Exception in thread "main" com.microsoft.sqlserver.jdbc.SQLServerException: Invalid Azure Key Vault key path specified: https://sunmsy.vault.azure.cn/keys/test/d03f8000d80645cb9d26756e290d2076.
	at com.microsoft.sqlserver.jdbc.SQLServerColumnEncryptionAzureKeyVaultProvider.ValidateNonEmptyAKVPath(SQLServerColumnEncryptionAzureKeyVaultProvider.java:456)
	at com.microsoft.sqlserver.jdbc.SQLServerColumnEncryptionAzureKeyVaultProvider.encryptColumnEncryptionKey(SQLServerColumnEncryptionAzureKeyVaultProvider.java:287)
	at com.sqlkeyvalut.demo.DemoApplication.createCEK(DemoApplication.java:148)
	at com.sqlkeyvalut.demo.DemoApplication.testAKV(DemoApplication.java:99)
	at com.sqlkeyvalut.demo.DemoApplication.main(DemoApplication.java:83)

**Comment by lilgreenbird** [comment]
hi @sunmsy

Please take a look at the documentation on [Using Always Encrypted with the JDBC driver](https://docs.microsoft.com/en-us/sql/connect/jdbc/using-always-encrypted-with-the-jdbc-driver?view=sql-server-2017#using-built-in-column-master-key-store-providers)

You can also find an Azure Key Vault Sample [here]
(https://docs.microsoft.com/en-us/sql/connect/jdbc/azure-key-vault-sample-version-7.0?view=sql-server-2017)

If you are still having issues, please share your code with us so we can investigate further.

**Comment by lilgreenbird** [comment]
the Azure key vault domain name is assumed to end with "vault.azure.net" which is wrong for Azure China or other areas. We will look into fixing this.

**Comment by sunmsy** [comment]
@lilgreenbird Thank you very much for your reply. If this bug is fixed, please let me know.

**Comment by lilgreenbird** [comment]
We haven't set a timeout for the next release set, but typically this will likely be within the next couple of months. This issue will remain open until we publish the fix, if you subscribe to notifications you will be notified when the fix is merged. In the meantime, if you require a workaround sooner, you can clone the repo and customize this by removing the following lines from SQLServerColumnEncryptionAzureKeyVaultProvider.java:
```450   // A valid URI.
451   // Check if it is pointing to AKV.
452   if (!parsedUri.getHost().toLowerCase(Locale.ENGLISH).endsWith(azureKeyVaultDomainName)) {
453       // Return an error indicating that the AKV url is invalid.
454       MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_AKVMasterKeyPathInvalid"));
455       Object[] msgArgs = {masterKeyPath};
456       throw new SQLServerException(null, form.format(msgArgs), null, 0, false);
457   }
```

**Comment by ulvii** [comment]
Hi @sunmsy ,
We fixed the issue in #1130 . I also attached the jars below, would you give them a try and let us know if your issue is resolved?
[PR1130.zip](https://github.com/microsoft/mssql-jdbc/files/3507316/PR1130.zip)

**Comment by ulvii** [comment]
Fixed in #1130

**Comment by sunmsy** [comment]
Thanks, It works!
